src/PrestaShopBundle/Resources/config/services/core/stock.yml

```diff
@@ -0,0 +1,6 @@
+services:
+    _defaults:
+        public: true
+
+    prestashop.core.stock.stock_manager:
+        class: 'PrestaShop\PrestaShop\Core\Stock\StockManager'
```

Comment:
I integrated this one so that it can be easily injected in handers/services
However it as built a long time ago and relies on `PrestaShop\PrestaShop\Adapter\ServiceLocator` and `PrestaShop\PrestaShop\Adapter\SymfonyContainer`

I think it would be worth having a clean refacto so that it is fully built by Symfony framework and DI, however it will probably be a BC since it might be used in core or in modules by just doing `new StockManager ()` At the very least it is used in `StockAvailable` like this `$stockManager = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Core\\Stock\\StockManager');`

So we maybe need to create a duplicate class with proper constructor and keep this one as deprecated

Reply:
>So we maybe need to create a duplicate class with proper constructor and keep this one as deprecated

I'm in favor of that 👍

Reply:
In **much** later PR then ^^